libs/external-db-mysql/src/mysql_schema_provider.ts

```diff
@@ -74,27 +76,38 @@ export default class SchemaProvider implements ISchemaProvider {
                          .catch( err => translateErrorCodes(err, collectionName) )
     }
 
-    async describeCollection(collectionName: string): Promise<ResponseField[]> {
-        const res = await this.query(`DESCRIBE ${escapeTable(collectionName)}`)
-                              .catch( err => translateErrorCodes(err, collectionName) )
-        return res.map((r: { Field: string; Type: string }) => ({ field: r.Field, type: r.Type }))
-                  .map( this.translateDbTypes.bind(this) )
+    async describeCollection(collectionName: string): Promise<Table> {
+        interface describeTableResponse {
+            Field: string,
+            Type: string,
+        }
+        
+        const res: describeTableResponse[] = await this.query(`DESCRIBE ${escapeTable(collectionName)}`)
+                                                       .catch( err => translateErrorCodes(err, collectionName) )
+        const fields = res.map(r => ({ field: r.Field, type: r.Type })).map(this.appendAdditionalRowDetails.bind(this))
+        return {
+            id: collectionName,
+            fields: fields as ResponseField[],
+            capabilities: this.collectionCapabilities(res.map(f => f.Field))
+        }
     }
 
-    translateDbTypes(row: ResponseField): ResponseField {
+    appendAdditionalRowDetails(row: ResponseField) {
         row.type = this.sqlSchemaTranslator.translateType(row.type)
+        row.capabilities = columnCapabilitiesFor(row.type)
         return row
     }
 
-    columnCapabilitiesFor(columnType: string): ColumnCapabilities {
-        return columnCapabilitiesFor(columnType)
+    translateDbTypes(row: ResponseField): ResponseField {
```

Comment:
could be private method

Reply:
Um, right now it's not possible because the translateDbTypes method appears in the ISchemaProvider interface as public

Reply:
On second thought, after this PR, we won't use this method anymore.
So I delete this method completely.